frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.tsx

```diff
@@ -70,30 +85,54 @@ export const AccessControlTab = ({ org, app }: AccessControlTabProps): ReactNode
         );
       }
       case 'success': {
-        const currentPartyTypesAllowed = appMetadata?.partyTypesAllowed ?? initialPartyTypes;
+        const isNoCheckboxesChecked = Object.values(appMetadata.partyTypesAllowed).every(
+          (value) => !value,
+        );
+        const areAllCheckboxesChecked = Object.values(appMetadata.partyTypesAllowed).every(
+          (value) => value,
+        );
+        const isSomeCheckboxesChecked = Object.values(appMetadata.partyTypesAllowed).some(
+          (value) => value,
+        );
         return (
-          <Checkbox.Group
-            legend={t('settings_modal.access_control_tab_checkbox_legend')}
-            size='small'
-            onChange={(newValues: string[]) => handleChange(newValues, currentPartyTypesAllowed)}
-            value={Object.keys(currentPartyTypesAllowed).filter(
-              (key) => currentPartyTypesAllowed[key],
-            )}
-          >
-            <Paragraph asChild size='small' short className={classes.checkboxParagraph}>
+          <>
+            <TabHeader text={t('settings_modal.access_control_tab_checkbox_legend_label')} />
+            <Paragraph size='medium'>
               <span>{t('settings_modal.access_control_tab_checkbox_description')}</span>
             </Paragraph>
-            {displayCheckboxes()}
-          </Checkbox.Group>
+            <SelectAllowedPartyTypes
+              t={t}
+              appMetadata={appMetadata}
+              isAllChecked={areAllCheckboxesChecked}
+              isSomeChecked={isSomeCheckboxesChecked}
+              isNoneChecked={isNoCheckboxesChecked}
+              handleTableHeaderCheckboxChange={handleTableHeaderCheckboxChange}
+              handleAllowedPartyTypeChange={handleAllowedPartyTypeChange}
+              getPartyTypesAllowedOptions={getPartyTypesAllowedOptions}
+            />
```

Comment:
I suggest that we only pass these:
```
<SelectAllowedPartyTypes
              partyTypesAllowed={appMetadata.partyTypesAllowed}
              handleTableHeaderCheckboxChange={handleTableHeaderCheckboxChange}
              handleAllowedPartyTypeChange={handleAllowedPartyTypeChange}
            />
```
The remaining we can deduce or fetch directly where it is used in the component 😊

Reply:
actually I have changed my mind, you can also move the methods; handleAllowedPartyTypeChange and handleTableHeaderCheckboxChange, and only pass the mutation `updateAppmetadata` down to the component. So the result is this:
 ```
 <SelectAllowedPartyTypes
              partyTypesAllowed={appMetadata.partyTypesAllowed}
              onUpdateAppMetadata={updateAppMetadata}
            />
```

Reply:
fixed

Reply:
I think we should continue passing at least appmetadata from above, since that above component checks if the metadata is fetched first, so it is not performance wise good, or needed, to fetch it again (even though it will be fetched from the cache)

UpdateAppMetadata can stay inside the component, as your code is currently, tho 😊 

That will also remove the need to mock the api call to getMetadata for. the SelectAllowedPartyTypes test 👍